packages/wonder-blocks-core/util/add-style.md

```diff
@@ -0,0 +1,167 @@
+Adds a `style` property to a component. A component can be either a custom React
+Component (e.g. Link, Button) or a DOM intrinsic (e.g. "div", "span"). This
```

Comment:
Given that we mention costume React Components here, I think it's worth highlighting when `addStyle` should not be used.  In particular, if we've defined a component ourselves, it's preferable to simply add a `style: StyleType` prop to the `Props` definition ourselves and avoid the `addStyle` property.  Also, in the case of React components from other packages, they may not handle `className` or `style` in which case wrapping them in `addStyle` won't do anything.

Reply:
Can you add an example of adding `style: StyleType` to the `Props` of a custom component and then show how we can pass the incoming `style` into an array of other styles we pass to a child component?